backend/tournesol/views/subsamples.py

```diff
@@ -0,0 +1,94 @@
+"""
+The sub-samples API return a sub-sample of entities compared by a user.
+
+A sub-sample is a small finite number of compared entities, ranked by the
+individual score computed for the poll's main criterion.
+"""
+import random
+
+import numpy as np
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from tournesol.models import ContributorRating
+from tournesol.serializers.subsample import SubSampleSerializer
+from tournesol.views.ratings import ContributorRatingQuerysetMixin
+
+# The number of ranked group used to divide the rated entities.
+DEFAULT_BUCKET_COUNT = 20
+
+
+class SubSamplesQuerysetMixin(ContributorRatingQuerysetMixin):
+    def get_queryset(self):
+        poll = self.poll_from_url
+
+        try:
+            sub_sample_size = int(self.request.query_params["ntile"])
+        except KeyError:
+            sub_sample_size = DEFAULT_BUCKET_COUNT
+
+        # Return a sub-sample of the user's ratings, without annotations so as
+        # not to affect performance.
+        all_ratings = (
+            ContributorRating.objects
+            .filter(
+                poll=poll,
+                user=self.request.user,
+                criteria_scores__criteria=poll.main_criteria
+            )
+            .order_by("-criteria_scores__score")
+            .values_list("entity_id")
```

Comment:
(very minor)
When selecting a single field, [`values_list()` ](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#values-list) accepts an argument `flat=True` to get a list of values, instead of a list of tuples. This would avoid using `rating[0]` on line 58, and may improve readability a bit.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1765/commits/385baec531bb09f9416fe398f00f5c4f13cf9f13 and https://github.com/tournesol-app/tournesol/pull/1765/commits/1e689640a8f398234b99428f53dbb50a62daab5f